homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -149,15 +149,51 @@ def _escape_decode(value: Any) -> Any:
     return value
 
 
-def _chat_message_convert(
-    message: conversation.Content | conversation.NativeContent[genai_types.ContentDict],
-) -> genai_types.ContentDict:
-    """Convert any native chat message for this agent to the native format."""
-    if message.role == "native":
-        return message.content
+def _create_google_tool_response_content(
+    content: list[conversation.ToolResultContent],
+) -> protos.Content:
+    """Create a Google tool response content."""
+    return protos.Content(
+        parts=[
+            protos.Part(
+                function_response=protos.FunctionResponse(
+                    name=tool_result.tool_name, response=tool_result.tool_result
+                )
+            )
+            for tool_result in content
+        ]
+    )
+
 
-    role = "model" if message.role == "assistant" else message.role
-    return {"role": role, "parts": message.content}
+def _convert_content(content: conversation.Content) -> genai_types.ContentDict:
+    """Convert HA content to Google content."""
+    assert content.role != "tool_result"
```

Comment:
My impression is that some of the other core members aren't aligned on using assert for typing like this in production code.  (and i'm not sure they've suggested an alternative, implicitly suggesting they want untested if statements and assertions for shouldn't happen cases)

Reply:
Yeah, I should probably change the typing on `content` param to fix this.